editor.planx.uk/src/pages/FlowEditor/lib/store/shared.ts

```diff
@@ -88,8 +90,8 @@ export const sharedStore: StateCreator<
     removeSessionIdSearchParam();
   },
 
-  setFlow({ id, flow, flowSlug }) {
-    this.setFlowNameFromSlug(flowSlug);
+  setFlow({ id, flow, flowSlug, flowName }) {
+    this.setFlowName(flowName);
     set({ id, flow, flowSlug });
```

Comment:
A couple questions: 
- Above on L37, you have `setFlowName: (flowSlug: string) => void;` - should the expected argument here actually be `flowName` based on how you're calling it later on L94? 
- If the expected argument is `flowName`, not `flowSlug`, could the call here be simplified to this ?
```
setFlow({ id, flow, flowSlug, flowName }) {
  set({ id, flow, flowSlug, flowName });
  get().initNavigationStore();
}
```